llvm/lib/Target/AIE/AIEMaxLatencyFinder.cpp

```diff
@@ -131,6 +129,46 @@ int MaxLatencyFinder::findEarliestRef(
   return Cycle;
 }
 
+/// Whether the SUnits in the DAG have know successor instructions. Those
+/// can be from successor MBBs, or from a fall-through timed region inserted
+/// by the AIEPostPipeliner.
+bool hasScheduledSuccessorInstructions(MachineBasicBlock::iterator RegionEnd,
+                                       MachineBasicBlock &MBB,
+                                       const AIEPostRASchedStrategy &Sched) {
+  const BlockState &BS = Sched.getInterBlock().getBlockState(&MBB);
+  return isBottomRegion(RegionEnd, MBB) &&
+         (!BS.BottomInsert.empty() || Sched.successorsAreScheduled(&MBB));
+}
+
+/// Return the list of scheduled instructions that will follow those in the DAG.
+SmallVector<ArrayRef<AIE::MachineBundle>, 2>
+getScheduledSuccessorInstructions(MachineBasicBlock &MBB,
+                                  const AIEPostRASchedStrategy &Sched) {
+  const BlockState &BS = Sched.getInterBlock().getBlockState(&MBB);
+  assert(!BS.BottomInsert.empty() || Sched.successorsAreScheduled(&MBB));
+
+  // The block has a timed sequence of instructions at the bottom, use
+  // it as the successor instructions.
+  if (!BS.BottomInsert.empty()) {
+    return {BS.BottomInsert};
+  }
+
+  // Otherwise, look at the top region for the successor MBBs.
+  SmallVector<ArrayRef<AIE::MachineBundle>, 2> ScheduledRegs;
+  for (MachineBasicBlock *SuccBB : MBB.successors()) {
+    auto &SBS = Sched.getInterBlock().getBlockState(SuccBB);
+    assert(SBS.isScheduled());
+    if (SBS.getRegions().empty()) {
```

Comment:
I am curious about how rare this can be. If this can frequent, we can take the successor's successor to reduce the pessimism.